obs/window-basic-main.cpp

```diff
@@ -4403,3 +4439,90 @@ void OBSBasic::on_actionLockPreview_triggered()
 	ui->preview->ToggleLocked();
 	ui->actionLockPreview->setChecked(ui->preview->Locked());
 }
+
+void OBSBasic::ToggleShowHide()
+{
+	if (isVisible()) {
+		config_set_string(App()->GlobalConfig(), "BasicWindow", "geometry",
+			saveGeometry().toBase64().constData());
+		showHide->setText(QTStr("Basic.SystemTray.Show"));
+		QTimer::singleShot(250, this, SLOT(hide()));
+		EnablePreviewDisplay(false);
+	} else {
+		showHide->setText(QTStr("Basic.SystemTray.Hide"));
+		QTimer::singleShot(250, this, SLOT(show()));
+		EnablePreviewDisplay(true);
+	}
+}
+
+void OBSBasic::SystemTrayInit() {
+	trayIcon = new QSystemTrayIcon(QIcon(":/res/images/obs.png"), 
+			this);
+	sysTrayStream = new QAction(QTStr("Basic.Main.StartStreaming"),
+			trayIcon);
+	sysTrayRecord = new QAction(QTStr("Basic.Main.StartRecording"), 
+			trayIcon);
+	showHide = new QAction(QTStr("Basic.SystemTray.Show"), 
+			trayIcon);
+	exit = new QAction(QTStr("Exit"), 
+			trayIcon);
+
+	connect(sysTrayStream, SIGNAL(triggered()), 
+			this, SLOT(on_streamButton_clicked()));
+	connect(sysTrayRecord, SIGNAL(triggered()), 
+			this, SLOT(on_recordButton_clicked()));	
+	connect(showHide, SIGNAL(triggered()), 
+			this, SLOT(ToggleShowHide()));
+	connect(exit, SIGNAL(triggered()), 
+			this, SLOT(SystemTrayExit()));
+
+	trayMenu = new QMenu;
+	trayMenu->addAction(sysTrayStream);
+	trayMenu->addAction(sysTrayRecord);
+	trayMenu->addAction(showHide);
+	trayMenu->addAction(exit);
+	trayIcon->setContextMenu(trayMenu);
+}
+
+void OBSBasic::SystemTrayExit()
+{
+	QTimer::singleShot(250, this, SLOT(show()));
+	close();
```

Comment:
The window position is still lost for me, when i close OBS using the exit menu entry. I'm not sure if this code makes sense. You try to make the window visible with a 250ms delay, but then you immediately close it. Maybe a better solution would be to remove the geometry saving from the destructor and execute it every time the window is hidden, which should happen on normal exit and when using "Hide" from the tray icon. Then you can close the hidden window without messing up the saved geometry.